### PR TITLE
Frontend cleanup

### PR DESCRIPTION
**Pull request by alex-ju**
This PR does the following in an attempt to code consistency:
- Update templates to only generate unique `id`s when mandatory and make use of rails view helpers to simplify the logic and preserve markup formatting
- Rename `dropdown-spec` to `toolbar-dropdown-spec` as it was missed when renaming the component
- Declare `eslint-env` consistently in spec files
- Fixed a wrongly formatted YAML line which is currently breaking the component-guide

### CODE REVIEW

File: app/views/components/_warning_summary.html.erb [hunk 1]
```diff
@@ -2,30 +2,27 @@
   title ||= false
   description ||= false
   items ||= []
-  id ||= "app-c-warning-summary-#{SecureRandom.hex(4)}"
+  id ||= id
```
Comment (kevindew) [inline app/views/components/_warning_summary.html.erb]: is this meant to be nil?

Comment (alex-ju) [reply app/views/components/_warning_summary.html.erb]: 🤦‍♂️ yep, good spot.

Comment (alex-ju) [reply app/views/components/_warning_summary.html.erb]: Updated

Comment (kevindew) [reply app/views/components/_warning_summary.html.erb]: Looks still here - did you push?

Comment (alex-ju) [reply app/views/components/_warning_summary.html.erb]: Yep, put the it seems like I got the same error in two files, now is fixed in both.

**Comment by benthorner** [comment]
Nice one @alex-ju !

**Pull request merged**